pkg/gardenlet/controller/seed/seed_control.go

```diff
@@ -414,3 +421,24 @@ func deleteBackupBucketInGarden(ctx context.Context, k8sGardenClient client.Clie
 
 	return client.IgnoreNotFound(k8sGardenClient.Delete(ctx, backupBucket))
 }
+
+// CleanupLegacyPriorityClasses deletes reversed-vpn-auth-server, fluent-bit, loki, istio-ingressgateway and istiod priority classes
+func CleanupLegacyPriorityClasses(ctx context.Context, seedClient client.Client) error {
+	var fns []flow.TaskFn
+	legacyPriorityClasses := sets.NewString([]string{"reversed-vpn-auth-server", "fluent-bit", "loki", "istio-ingressgateway", "istiod"}...)
+
+	priorityClasses := &schedulingv1.PriorityClassList{}
+	if err := seedClient.List(ctx, priorityClasses); err != nil {
```

Comment:
Can you avoid this listing and just delete with ignoring not found? Listing can be quite expensive.